cardinal/util.go

```diff
@@ -48,3 +52,34 @@ func isFatalError(err error) bool {
 	}
 	return true
 }
+
+func GetMessage[In any, Out any](wCtx Context) (*MessageType[In, Out], error) {
+	var msg MessageType[In, Out]
+	msgType := reflect.TypeOf(msg)
+	tempRes, ok := wCtx.getMessageByType(msgType)
+	if !ok {
+		return nil, eris.Errorf("Could not find %q, Message may not be registered.", msg.Name())
+	}
+	var _ types.Message = &msg
+	res, ok := tempRes.(*MessageType[In, Out])
+	if !ok {
+		return &msg, eris.New("wrong type")
+	}
+	return res, nil
+}
+
+func GetTransactionReceiptsForTick(wCtx Context, tick uint64) ([]receipt.Receipt, error) {
+	ctx, ok := wCtx.(*worldContext)
+	if !ok {
+		return nil, eris.New("error in test type assertion.")
+	}
+	return ctx.world.GetTransactionReceiptsForTick(tick)
+}
+
+func GetStoreManagerFromContext(wCtx Context) gamestate.Manager {
+	return wCtx.storeManager()
+}
+
+func GetComponentByNameFromContext(wCtx Context, name string) (types.ComponentMetadata, error) {
```

Comment:
I saw this is only used in `search_test.go` and I don't understand why you need this and `GetStoreManagerFromContext` instead of just doing `cardinal.SetComponent`

Reply:
When a method gets privatized and is used in a test I just dumbly wrapped it in a public function so the functionality remained the same. That's what's going on here. The test was previously using an internal method so I kept the functionality the same. That's what your seeing here with some of these public functions. 

Of course the method you mentioned here is better. I'll switch it.